OA-PASS/pass-docker
# Update to deposit-services-0.0.5

* Upgrades to deposit-services-0.0.5-0.0.2 ([release notes](https://github.com/OA-PASS/nihms-submission/releases/tag/deposit-services-0.0.5))
    * primarily to support the customization of queue names using environment variables `PASS_DEPOSIT_QUEUE_DEPOSIT_NAME` and `PASS_DEPOSIT_QUEUE_SUBMISSION_NAME`
* Places the NIH `ftpserver` on the backend network so it is reachable from the `deposit` container in a `docker-machine` environment
* Removes `SPRING_ACTIVEMQ_BROKER_URL` from `.env` as this release of deposit services uses the shared `ACTIVEMQ_BROKER_URI` variable.

## To test

### Docker startup
1. Clear everything out (`docker-compose down -v`)
2. Pull images (`docker-compose pull`)
3. Start 'er up (`docker-compose up -d`)
4. Tail the logs for the `deposit` container, (`docker logs -f deposit`)
    * wait to see `Fedora repository is up at 'http://192.168.99.100:8080/fcrepo/rest/'`

### Ember interaction
1. Go to `https://pass` and login as `incomplete-nih-user`, password `moo`
2. Go to the Grants tab, and select an NIH grant, for example `Profiling Cardiovascular Events and Biomarkers in the Very Old to Improve Personalized Approaches for the Prevention of Cardiac and Vascular Disease`
3. Go through the submission workflow accepting defaults, being sure to accept the JScholarship deposit agreement
    * I use DOI `10.1021/acsmedchemlett.7b00397`
4. Upload at least one file.
5. At this point, feel free to clear out your `deposit` scrollback buffer.
6. Click `Submit`

The `deposit` container should light up.  Wait for (or search for) messages like: `Updating deposit http://192.168.99.100:8080/fcrepo/rest/deposits/ba/d9/89/68/bad98968-e118-458e-8cd7-2143b48fb829`.  

### Verification
1. There should be two deposits (that is, the search for `Updating deposit...` should hit twice, and each `Deposit` URI should be unique).
2. For each `Deposit` resource, load it in your browser.
3. Determine if the `Deposit` is for `PubMed Central` or `JScholarship` by navigating to the `Repository`
4. For `Deposit`s to `PubMed Central`
    * Examine the `Deposit` `status` field: should be `submitted`
    * Note that in the past we have discussed what the deposit status of NIH deposits ought to be.  Past discussion (IIRC) decided that *deposit services* would update this to `accepted`, although that is really not the correct thing to do.  I'll probably open up a GitHub issue for this, so we can document the rationale and future path for deposit services' treatment of NIH deposits.
5. For `Deposit`s to `JScholarship`
    * Verify the `depositStatusRef` is not `null`.
    * Navigate to the `RepositoryCopy`, verify its status is `complete`
    * Copy and paste the `externalIds` value into your browser.  The DSpace Item page should be presented, with accurate metadata (including accurate file names associated with the item).   
    * Because the files are put in a subdirectory named `data/` in the package, they appear underneath the URL path `data/`
    * Clicking on `Full Item Record` should show sane values, including Contributors, URIs and DOIs
6. * Follow the reference to the `Submission` URI (note each `Deposit` should point to the same `Submission`)
    * Check the `Submission` aggregate deposit status to be `in-progress`
    * This is because the `PubMed Central` `Deposit` is `submitted` and not `accepted`
